New feature: CLM filter templates

cbbayburt opened the pull request:
# Description

New feature: CLM filter templates

# Target branches

- [x] Master (Default)
- [x] Manager-4.2

# Links

Feature PR: https://github.com/uyuni-project/uyuni/pull/3561

keichwa commented on modules/administration/pages/content-lifecycle-examples.adoc:
```diff
@@ -184,6 +184,7 @@ The new environment contains all the kernel patches up to the version you instal
 NOTE: All kernel patches with higher kernel versions are removed.
 Live patching kernels remain available as long as they are not the first in a series.
 
+NOTE: This procedure can be automated using a filter template. For more information on how to apply a live patching filter template, see xref:administration:content-lifecycle.adoc#filter-templates[].
```
Maybe, we can turn this into a formal note (it's new, thus it may deserve more visibility):
```suggestion
[NOTE]
====
This procedure can be automated using a filter template.
For more information on how to apply a live patching filter template, see xref:administration:content-lifecycle.adoc#filter-templates[].
====
```

keichwa commented on modules/administration/pages/content-lifecycle.adoc:
```diff
@@ -113,6 +113,46 @@ If there is no previous environment, it is left empty until the project sources
+== Filter Templates
+
+To help with creating filters for some common scenarios, {productname} offers filter templates.
+When applied, these templates help create a set of pre-defined filters, tailored for a specific use case.
```
```suggestion
When applied, these templates help creating a set of filters in advance, tailored for a specific use case.
```

keichwa commented on modules/administration/pages/content-lifecycle.adoc:
```diff
@@ -113,6 +113,46 @@ If there is no previous environment, it is left empty until the project sources
+
+When applied, this template creates two filters required to achieve this behavior:
+
+* Deny patches that contain `kernel-default` package greater than a specific kernel version
```
I think asciidoc wants two tics...
```suggestion
* Deny patches that contain ``kernel-default`` package greater than a specific kernel version
```

keichwa commented on modules/administration/pages/content-lifecycle.adoc:
```diff
@@ -113,6 +113,46 @@ If there is no previous environment, it is left empty until the project sources
+When applied, this template creates two filters required to achieve this behavior:
+
+* Deny patches that contain `kernel-default` package greater than a specific kernel version
+* Deny patches that contain `reboot_suggested` keyword
```
```suggestion
* Deny patches that contain ``reboot_suggested`` keyword
```

keichwa commented on modules/administration/pages/content-lifecycle.adoc:
```diff
@@ -113,6 +113,46 @@ If there is no previous environment, it is left empty until the project sources
+
+[IMPORTANT]
+====
+The second "catch-all" filter is a safeguard to make sure the assigned clients are never required a restart in the future. You can choose not to attach it to the project if this behavior is not desired as it might block other important updates.
```
We have this strange policy (I think it help adoc editors): New sentence = new line
```suggestion
The second "catch-all" filter is a safeguard to make sure the assigned clients are never requiring a restart in the future.
You can choose not to attach it to the project if this behavior is not desired because it might block other important updates.
```

keichwa commented on modules/administration/pages/content-lifecycle.adoc:
```diff
@@ -113,6 +113,46 @@ If there is no previous environment, it is left empty until the project sources
+. Click btn:[Save] to create the filters.
+. Navigate to menu:Content Lifecycle[Projects] and select your project.
+. Click btn:[Attach/Detach Filters].
+. Select the 2 filters that have the specified prefix, and click btn:[Save].
```
```suggestion
. Select the two filters that have the specified prefix, and click btn:[Save].
```

keichwa reviewed (approved):
Thank your Can! Some minor suggestions.

@0rnela , can you help with English phrases? I'm not sure about my proposals!

cbbayburt commented:
Thanks for the quick review @keichwa ! I've addressed your comments in a separate commit. Let's wait for @0rnela 's approval and then we can merge it.
